### PR TITLE
Problem: can't run CLI related tests on kubernetes

### PR DESCRIPTION
**Pull request by dkliban**
Solution: add ability to specify 'kubectl' as shell transport

### CODE REVIEW

File: pulp_smash/cli.py [hunk 1]
```diff
@@ -232,6 +233,11 @@ def machine(self):
 
             if transport == "local":
                 self._machine = plumbum.machines.local
+            elif transport == "kubectl":
+                self._machine = plumbum.machines.local
+                chain = self._machine["kubectl"]["get", "pods"] | \
+                        self._machine["grep"]["-E", "pulp-api-(\w+)-(\w+)"]
+                self._podname = chain().split(" ")[0]
```
Comment (mikedep333) [inline pulp_smash/cli.py]: Are you trying to get only the name of the pod? If so, just add "-o" to grep.

File: pulp_smash/cli.py [hunk 2]
```diff
@@ -276,6 +282,9 @@ def run(self, args, sudo=False, **kwargs):
         kwargs.setdefault("retcode")
         logger.debug("Running %s cmd (sudo:%s) - %s", args, sudo, kwargs)
 
+        if self._podname:
+            args = ("kubectl", "exec", self._podname) + tuple(args)
```
Comment (mikedep333) [inline pulp_smash/cli.py]: We need -- (dash dash) at the end also.

Comment (mikedep333) [inline pulp_smash/cli.py]: We need -- (dash dash) at the end also.

So that we can run commands with arguments.

File: pulp_smash/cli.py (hunk 1 above)
Comment (dkliban) [reply pulp_smash/cli.py]: yep, i added teh -o. thank you!

File: pulp_smash/cli.py [hunk 3]
```diff
@@ -276,6 +282,9 @@ def run(self, args, sudo=False, **kwargs):
         kwargs.setdefault("retcode")
         logger.debug("Running %s cmd (sudo:%s) - %s", args, sudo, kwargs)
 
+        if self._podname:
+            args = ("kubectl", "exec", self._podname) + tuple(args) + ("--",)
```
Comment (mikedep333) [inline pulp_smash/cli.py]: The -- needs to become before the command you actually run.

**Pull request merged**